### PR TITLE
Change order of items by dragging and dropping the items

### PR DESCRIPTION
**Pull request by SjoerdB93**
The proof-of-concept is there, and it works quite well actually:

[Screencast from 2023-04-02 21-16-09.webm](https://user-images.githubusercontent.com/68477016/229374154-6d14f3e0-cba8-40ee-9d22-40e7b9ab04f1.webm)

TODO:

- [X] Fix UI (The dragging preview shouldn't be the ID, ideally it should show the row but at least it should show item name)
- [X] Code (check if code is OK enough, but at least fix linting, am aware some lines are too long atm)

### CODE REVIEW

File: src/utilities.py [hunk 1]
```diff
@@ -23,6 +23,17 @@ def sig_fig_round(number, digits):
     power = "{:e}".format(float(number)).split("e")[1]
     return round(float(number), -(int(power) - digits + 1))
 
+def swap_key_positions(d, key1, key2):
```
Comment (cmkohnen) [inline src/utilities.py:27]: 'd' should probably be 'dict' or 'dictionary'

File: src/item_box.py [hunk 2]
```diff
@@ -33,6 +34,36 @@ def __init__(self, parent, item):
...
+        self.parent.item_menu = utilities.swap_key_positions(
+        self.parent.item_menu, drop_target.key, value)
+        for key, item in self.parent.item_menu.items():
+            self.parent.main_window.list_box.remove(item)
+            self.parent.main_window.list_box.append(item)
+        graphs.reload(self.parent)
+
+
+    def on_dnd_prepare(self, drag_source, x, y):
+        data = self.item.key
```
Comment (cmkohnen) [inline src/item_box.py:62]: would `utilities.shorten_label(self.item.name)` suffice to change to the tooltip to the actual name or is the key required for this to function?

**Review by cmkohnen — COMMENTED** [review]
Looks good!
Apart from the flake comments, this looks clean

File: src/utilities.py (hunk 1 above)
Comment (SjoerdB93) [reply src/utilities.py]: Yes that's probably better

**Review by cmkohnen — APPROVED** [review]

**Pull request merged**